core/api/src/main/java/com/blazebit/persistence/PagedArrayList.java

```diff
@@ -40,21 +38,38 @@
     private final int firstResult;
     private final int maxResults;
 
-    public PagedListImpl(KeysetPage keyset, long totalSize, int firstResult, int maxResults) {
+    /**
+     * Constructs a new empty paged array list.
+     *
+     * @param keyset      The keyset page for this page
+     * @param totalSize   The total size of the result
+     * @param firstResult The first result index within the overall result
+     * @param maxResults  The maximum result count for a page
+     */
+    public PagedArrayList(KeysetPage keyset, long totalSize, int firstResult, int maxResults) {
         this.keyset = keyset;
         this.totalSize = totalSize;
         this.page = (int) Math.floor((firstResult == -1 ? 0 : firstResult) * 1d / maxResults) + 1;
-        this.totalPages = (int) Math.ceil(totalSize * 1d / maxResults);
+        this.totalPages = totalSize < 1 ? (int) totalSize : (int) Math.ceil(totalSize * 1d / maxResults);
```

Comment:
1. `totalSize * 1d / maxResults` -> `totalSize / (float) maxResults`
2. I don't get the added logic

Reply:
Not sure if the change in 1. has the same sematics/precision. I'll leave that as it is.